js/packages/components/main/Permissions.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { useEffect, useRef } from 'react'
+import { Text, TouchableOpacity, Platform, View, AppState } from 'react-native'
+import LottieView from 'lottie-react-native'
+import { useTranslation } from 'react-i18next'
+import {
+	requestNotifications,
+	request,
+	PERMISSIONS,
+	RESULTS,
+	openSettings,
+} from 'react-native-permissions'
+
+import { useStyles } from '@berty-tech/styles'
+import { PersistentOptionsKeys, useMsgrContext } from '@berty-tech/store/context'
+
+import audioLottie from '@berty-tech/assets/audio-lottie.json'
+import cameraLottie from '@berty-tech/assets/camera-lottie.json'
+import notificationLottie from '@berty-tech/assets/notification-lottie.json'
+import p2pLottie from '@berty-tech/assets/p2p-lottie.json'
+import { checkPermissions } from '../utils'
+
+const animations = {
+	audio: audioLottie,
+	camera: cameraLottie,
+	notification: notificationLottie,
+	p2p: p2pLottie,
+}
+
+export const Permissions: React.FC<{}> = (props) => {
+	const appState = useRef(AppState.currentState)
+	const [{ text, border, background }] = useStyles()
+	const { t }: { t: any } = useTranslation()
+	const { persistentOptions, setPersistentOption, createNewAccount } = useMsgrContext()
+	const {
+		permissionType,
+		permissionStatus,
+		navigateNext,
+		createNewAccount: isToCreateNewAccount,
+	} = props?.route?.params
+
+	const handleOnComplete = async () => {
+		if (isToCreateNewAccount) {
+			await createNewAccount()
+		}
+		if (navigateNext) {
+			props.navigation.navigate(navigateNext, {})
+		} else {
+			props.navigation.goBack()
+		}
+	}
+
+	const handleAppStateChange = async (nextAppState: string) => {
+		if (appState.current.match(/inactive|background/) && nextAppState === 'active') {
```

Comment:
could you add a comment with the rationale for checking permissions on AppState change pls? can't figure it out

Reply:
Ok. If the permission is denied, then we will show user "Go to Setting" button to let them enable permission from Settings. After enabling it from settings, user will be back to our app.At that moment, we will check if permission is enabled or not using AppState change; if enabled we will navigate back. If not user will be on same screen. I hope it's clear now.